api/src/common/decorators_api_views.py

```diff
@@ -0,0 +1,18 @@
+from django.utils.decorators import method_decorator
+from drf_yasg.utils import swagger_auto_schema
+
+# common decorators applied to all model view set classes
+def viewsets(tag):
+    print ('tag name is: ' + tag)
+
+    @method_decorator(name="list", decorator=swagger_auto_schema(tags=[tag]))
+    @method_decorator(name="create", decorator=swagger_auto_schema(tags=[tag]))
+    @method_decorator(name="retrieve", decorator=swagger_auto_schema(tags=[tag]))
+    @method_decorator(name="update", decorator=swagger_auto_schema(tags=[tag]))
+    @method_decorator(name="partial_update", decorator=swagger_auto_schema(tags=[tag]))
+    @method_decorator(name="destroy", decorator=swagger_auto_schema(tags=[tag]))
+    def wrapper(Cls):
+        print(Cls)
+        print(tag)
+        return Cls
+    return wrapper
```

Comment:
We should make sure the file ends with a new line.

Reply:
We can probably simplify this into a 1 line return, where we return a lambda expression like `return lambda cls : cls`. This will return a function that takes in a class and returns a class.

Reply:
using that gives invalid syntax....reading about it online says, lambda can only return expressions, not functions or classes

Reply:
Ah my bad lol 😅 I guess my python knowledge is lacking 😂 . Thanks for the info!